components/input/Input.js

```diff
@@ -0,0 +1,196 @@
+import React, { Component, createElement } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import {
+  IconChevronUpSmallOutline,
+  IconChevronDownSmallOutline,
+  IconWarningBadgedSmallFilled,
+} from '@teamleader/ui-icons';
+import InputMetaPropTypes from './InputMetaPropTypes';
+import Box from '../box';
+import Counter from '../counter';
+import { TextSmall } from '../typography';
+import theme from './theme.css';
+
+export default class Input extends Component {
+  static propTypes = {
+    bold: PropTypes.bool,
+    className: PropTypes.string,
+    counter: PropTypes.number,
+    disabled: PropTypes.bool,
+    helpText: PropTypes.string,
+    icon: PropTypes.oneOfType([PropTypes.func, PropTypes.element]),
+    iconPlacement: PropTypes.oneOf(['left', 'right']),
+    id: PropTypes.string,
+    inverse: PropTypes.bool,
+    meta: InputMetaPropTypes,
+    onBlur: PropTypes.func,
+    onChange: PropTypes.func,
+    onFocus: PropTypes.func,
+    placeholder: PropTypes.string,
+    readOnly: PropTypes.bool,
+    size: PropTypes.oneOf(['small', 'medium', 'large']),
+    step: PropTypes.number,
+    type: PropTypes.string,
+    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+  };
+
+  static defaultProps = {
+    iconPlacement: 'left',
+    inverse: false,
+    disabled: false,
+    placeholder: '',
+    readOnly: false,
+    size: 'medium',
+    step: 1,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.isNumberInput = Boolean(props.type === 'number');
+    this.handleChange = ::this.onChange;
+    this.handleIncreaseValue = ::this.increaseValue;
+    this.handleDecreaseValue = ::this.decreaseValue;
+
+    this.state = {
+      value: this.isNumberInput ? Number(props.value) || '' : props.value || '',
+    };
+  }
+
+  componentWillUpdate(props, state) {
```

Comment:
PP: Props is not being used here, it can be removed

Reply:
Not sure if this comment is still about the code that I see, but props is clearly used in this function?

 :ok: by @lowiebenoot on 13/02/2018 13:53:49